### DISCUSSION TITLE
Did anyone configure devcontainers to work well together with playwright?

### DISCUSSION THREAD

**Discussion by alexbjorlig** [devcontainers/community#147] [Q&A]
It would be awesome if I could debug browser tests with `playwright test --debug`, and it opened up my host chrome browser?

**Comment by ykawakamy** [comment]
I have tried port forwarding, X11 forwarding, and desktop-lite devcontainer feature.

### port forwarding
The idea is to expose the test target to the host via port forwarding and run playwright on host.
This is the easiest, but you will be running playwright outside of devcontainer.

### x11 forwarding
on the devcontainer, need to run `npx playwright --with-deps`, and set the `DISPLAY` environment variable  .
And run the X11 server (e.g. vcxsrv) on  host.

I had difficulty setting the DISPLAY environment variable properly.

In my case, in devcontainer.json, `“containerEnv”:"(IP address not 127.0.0.1 of host):0.0”`

### Desktop-lite feature
I can launch the browser on devcontainer, but on a virtual desktop.
[desktop lite demo](https://github.com/ykawakamy/playwright-on-desktoplite)

-- 
sry, I'm not good at English, using deepl.
